### DISCUSSION TITLE
Generated File Variable Names do not match

### DISCUSSION THREAD

**Discussion by erlkonig26** [runelite/runelite#17530] [Q&A] [answered]
Hello.
Variable names in ComponentID and InterfaceID classes, which are being generated during maven install, are coming like :
"İNVENTORY" instead of "INVENTORY".
Because of this, my IDE can't resolve them and I can't compile the project.

Here are two screenshots which I think can give you an insight:

![variable_names_1](https://github.com/runelite/runelite/assets/161385682/a5b24950-ff35-4aa7-8a53-e560541b7752)

![variable_names_2](https://github.com/runelite/runelite/assets/161385682/25fc153c-f3b2-4636-a5d9-a1dd49ecc4c4)

Note: My system language is turkish, not us_en.

Has anyone experienced this problem before, or does anyone know why this happens and how I can solve this?

Thank you in advance.

**Comment by Nightfirecat** [comment]
What is your system language? I assume, not us_en

Chances are your system is capitalizing `i` to `İ` in ComponentMojo, where these classes are being generated: https://github.com/runelite/runelite/blob/a4e9071247a99cc2cc4946207a7fc61b0581a273/runelite-maven-plugin/src/main/java/net/runelite/mvn/ComponentMojo.java

**Reply by erlkonig26** [reply]
My system language is Turkish. Sorry I forgot to mention it in original post.
Is it allowed for me to change the codes in ComponentMo.java to have the variable names correct?
Or I will try to change my system language to us_en.

**Reply by Nightfirecat** [reply]
We'll look at fixing this. For the time being, you should be able to pass `Locale.US` as an argument to all cases where `.toUpperCase()` is called in that file

**Reply by erlkonig26** [reply]
In the meantime, I tried to change my system language but my Windows license is "Windows 11 CoreSingleLanguage".
So thank you so much, until it gets fixed, I will be using `Locale.US` argument as you said.

**Comment by erlkonig26** [comment]
I managed to get the variable names correctly.
Then I got Test failures with some plugins. I am listing them below:

- Test set: net.runelite.client.plugins.chatcommands.***ChatCommandsPluginTest***
(org.junit.ComparisonFailure: expected:<0:03[.]60> but was:<0:03[,]60>
	at net.runelite.client.plugins.chatcommands.ChatCommandsPluginTest.testSecondsToTimeString(ChatCommandsPluginTest.java:1072))

- Test set: net.runelite.client.plugins.emojis.***EmojiPluginTest***
(java.lang.IllegalArgumentException: slıght_smıle.png)

- Test set: net.runelite.client.plugins.menuentryswapper.***MenuEntrySwapperPluginTest***
(Wanted but not invoked:
client.setMenuEntries(<Capturing argument>);
-> at net.runelite.client.plugins.menuentryswapper.MenuEntrySwapperPluginTest.testBirdhouse(MenuEntrySwapperPluginTest.java:335))

**SOLUTION**
I have solved them by adding these lines into these files:

- ChatCommandsPlugin.java
```
@Override
	public void startUp()
	{
		Locale.setDefault(Locale.US);
...
}
```

- Emoji.java
```
BufferedImage loadImage()
	{
		return ImageUtil.loadImageResource(getClass(), this.name().toLowerCase(Locale.US) + ".png");
	}
```


- MenuEntrySwapperPlugin.java
```
@Override
	public void startUp()
	{
		Locale.setDefault(Locale.US);
		setupSwaps();
		removeOldSwaps();
	}
```


I wanted to add these so anyone who researchs ***variable names*** or something like that, can find here and see these solutions as well.

It would be really great if you can fix localization problems like these.
Thank you.

**Reply by cdfisher** [reply]
You should be able to undo those changes and just run the Maven build with `clean install -DskipTests -U` to resolve this.

**Comment by erlkonig26** [comment]
> You should be able to undo those changes and just run the Maven build with `clean install -DskipTests -U` to resolve this.

Will the game work fine? I thought there would be errors with those plugins.
I am going to try and edit my answer.

**Reply by cdfisher** [reply]
It should be fine. There are a couple situations where people's system settings can cause tests to fail and that's generally the solution.

**Reply by erlkonig26** [reply]
Yes, it works. But I still need to add `Locale.US` parameter to `toUpperCase()` functions in `ComponentMojo.java` because it generates the variable names. I rollbacked other changes I made.

**Comment by Adam-** [comment, marked as answer]
This has been fixed in 5623b498dc0406317106b622df4f325719c22a45

**Reply by erlkonig26** [reply]
Thank you.
